nni/compression/pytorch/compressor.py

```diff
@@ -887,28 +922,39 @@ def quant_backward(tensor, grad_output, quant_type, scale, zero_point, qmin, qma
     def forward(ctx, tensor, quant_type, wrapper, input_tensor=None, **kwargs):
         output = quantize_helper(tensor, quant_type, wrapper, input_tensor, **kwargs)
 
-        bits = QuantGrad.get_bits_length(wrapper.config, QType_Dict[quant_type])
-        qmin, qmax = torch.Tensor([0]).to(tensor.device), torch.Tensor([(1 << bits) - 1]).to(tensor.device)
-        if hasattr(wrapper.module, 'scale') and hasattr(wrapper.module, 'zero_point'):
+        if hasattr(wrapper.module, "layer_quant_setting"):
+            layer_quant_setting = wrapper.module.layer_quant_setting
+            qmin, qmax = getattr(layer_quant_setting, quant_type).get_qmin_qmax()
+        else:
+            # todo: when dtype/scheme customization is ready for all quantizers, remove this
+            bits = QuantGrad.get_bits_length(wrapper.config, quant_type)
+            qmin, qmax = 0, (1 << bits) - 1
+
+        scale_name, zero_point_name = quant_type.type_to_scale_zero_point_name()
```

Comment:
It is strange that we get scale_name and zero_point_name in this way while we register them in module using like 'weight_scale' and `weight_zero_point` directly. So what is the meaning of **type_to_scale_zero_point_name()**?

Reply:
Different `quant_type`s correspond to different scale&zero_point names, e.g. `weight_scale`, `input_scale` and `output_scale`. In order to getting them, we must map quantization types to the scale&zero point names. `type_to_scale_zero_point_name` is just for the code simplicity, do you have better ideas about this?